TR2RandomizerCore/Randomizers/ItemRandomizer.cs

```diff
@@ -335,6 +335,36 @@ private void RepositionItems(List<Location> ItemLocs)
                         }
                     }
                 }
+
+                if (Difficulty == ItemDifficulty.OneLimit)
+                {
+                    List<TR2Entities> oneOfEachType = new List<TR2Entities>();
+                    List<TR2Entity> allEntities = _levelInstance.Data.Entities.ToList();
+                    List<TR2Entity> toRemove = new List<TR2Entity>();
+
+                    // look for extra utility/ammo items and remove them
+                    foreach (TR2Entity ent in allEntities)
+                    {
+                        TR2Entities eType = (TR2Entities)ent.TypeID;
+                        if (TR2EntityUtilities.IsUtilityType(eType) ||
+                            TR2EntityUtilities.IsGunType(eType))
+                        {
+                            if (oneOfEachType.Contains(eType))
+                            {
+                                toRemove.Add(ent);
+                            }
+                            else
+                                oneOfEachType.Add((TR2Entities)ent.TypeID);
+                        }
+                    }
+                    foreach (TR2Entity ent in toRemove)
+                    {
+                        allEntities.Remove(ent);
+                    }
+
+                    _levelInstance.Data.Entities = allEntities.ToArray();
```

Comment:
As @lahm86 mentioned in discord, the removal of entities could have an effect on the entity IDs for triggering etc.